File: src/runtime/cs/std/os_path_native.cs
using System.IO;

namespace Pytra.CsModule
{
    // Generated std/os_path.cs delegates host bindings through this native seam.
    public static class os_path_native
    {
        public static string join(string a, string b)
        {
            return Path.Combine(a, b).Replace('\\', '/');
        }

        public static string dirname(string p)
        {
            return (Path.GetDirectoryName(p) ?? "").Replace('\\', '/');
        }

        public static string basename(string p)
        {
            return Path.GetFileName(p);
        }

        public static (string, string) splitext(string p)
        {
            string ext = Path.GetExtension(p);
            string stem = p.Substring(0, p.Length - ext.Length);
            return (stem, ext);
        }

        public static string abspath(string p)
        {
            return Path.GetFullPath(p).Replace('\\', '/');
        }

        public static bool exists(string p)
        {
            return File.Exists(p) || Directory.Exists(p);
        }
    }
}

File: src/runtime/julia/std/os_path_native.jl
# Native implementation for pytra.std.os_path
# Generated std/os_path.jl delegates host bindings through this native seam.

module __OsPathNative
    function abspath(p)
        return Base.abspath(p)
    end
    function basename(p)
        return Base.basename(p)
    end
    function dirname(p)
        return Base.dirname(p)
    end
    function exists(p)
        return Base.ispath(p)
    end
    function join(parts...)
        return Base.joinpath(parts...)
    end
    function splitext(p)
        root, ext = Base.splitext(p)
        return (root, ext)
    end
end
